### PR TITLE
Versioning

### PR DESCRIPTION
**Pull request by ykruchko**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ buildscript {
         maven { url "https://repo.spring.io/plugins-release" }
     }
     dependencies {
-        //jhipster-needle-gradle-buildscript-dependency - JHipster will add additional gradle build script plugins here
```
Comment (amedvedchuk) [inline build.gradle:11]: @ykruchko do not delete jhipser comment.

File: build.gradle [hunk 2]
```diff
@@ -368,3 +368,32 @@ task getProjectVersion {
...
 }
 
+import org.apache.commons.configuration.PropertiesConfiguration
+
+task incrementProjectVersion {
+    doLast{
+        PropertiesConfiguration props = new PropertiesConfiguration('gradle.properties')
+        String version = props.getProperty("version")
+        println "old version: " + version
+        String newVersion = version.tokenize(".")[0].toString() + "." + version.tokenize(".")[1].toString() + "." + (version.tokenize(".")[2].toInteger() + 1).toString()
```
Comment (amedvedchuk) [inline build.gradle]: Yoy do not need to tokenize version several times.
you can declare another array variable:
```
String [] vTokens = version.tokenize(".")
String newVersion = "${vTokens[0]}.${vTokens[1]}.${vTokens[2].toInteger() + 1}".toString()
```

File: build.gradle [hunk 3]
```diff
@@ -368,3 +368,32 @@ task getProjectVersion {
...
+}
+
+task decrementProjectVersion(dependsOn: configurations.compile) {
+    doLast{
+        PropertiesConfiguration props = new PropertiesConfiguration('gradle.properties')
+        String version = props.getProperty("version")
+        println "old version: " + version
+        String newVersion = version
+        if(version.tokenize(".")[2].toInteger() != 0){
+            newVersion = version.tokenize(".")[0].toString() + "." + version.tokenize(".")[1].toString() + "." + (version.tokenize(".")[2].toInteger() - 1).toString()
```
Comment (amedvedchuk) [inline build.gradle]: the same comment as above

File: deploy/env/balance-app.env [hunk 4]
```diff
@@ -0,0 +1,10 @@
+SPRING_PROFILES_ACTIVE=prod,swagger
+SPRING_CLOUD_CONSUL_HOST=consul
+SPRING_CLOUD_CONSUL_PORT=8500
+JHIPSTER_SLEEP=0
+SPRING_KAFKA_BOOTSTRAP_SERVERS=kafka:9092
+XMX=426m
+SPRING_DATASOURCE_HIKARI_MAXIMUM_POOL_SIZE=10
+SPRING_DATASOURCE_USERNAME=postgres
+SPRING_DATASOURCE_URL=jdbc:postgresql://postgres:5432/balance
+TEST=test
```
Comment (amedvedchuk) [inline deploy/env/balance-app.env]: do we need this TEST var here?

File: .github/workflows/autoincrement-project-version.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    steps:
+    - uses: actions/checkout@master
+      with:
+        ref: ${{ github.head_ref }}
+          
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: 8
```
Comment (amedvedchuk) [inline .github/workflows/autoincrement-project-version.yml:24]: should not we use java 11 instead of 8?

**Review by amedvedchuk — APPROVED** [review]

**Review by ishkurko — APPROVED** [review]

**Pull request merged**